utils/github_adapter.py

```diff
@@ -0,0 +1,232 @@
+from utils.github_api import GithubAPI
+import aiohttp, sys,os,httpx
+from utils.jwt_generator import GenerateJWT
+from events.ticketFeedbackHandler import TicketFeedbackHandler
+from utils.logging_file import logger
+
+
+
+class GithubAdapter:
+    def __init__(self):
+        return
+    
+    async def get_github_data(code):
+        try:
+            github_url_for_access_token = 'https://github.com/login/oauth/access_token'
+            data = {
+                "client_id": os.getenv("GITHUB_CLIENT_ID"),
+                "client_secret": os.getenv("GITHUB_CLIENT_SECRET"),
+                "code": code
+            }
+            headers = {
+                "Accept": "application/json"
+            }
+
+            async with aiohttp.ClientSession() as session:
+                async with session.post(github_url_for_access_token, data=data, headers=headers) as response:
+                    r = await response.json()                    
+                    return r
+                
+        except Exception as e:
```

Comment:
Have we thought through exception handling and logging?
If, for example, this endpoint starts failing, we'll stop receiving github details of individuals registering through the discord flow.

We should probably raise an alert for that, for example on a discord webhook on the testing server/samagra server?

Reply:
Im not clear about this comment... @KDwevedi  For now i raise an exception from except block